pkg/collector/check/jmx.go

```diff
@@ -13,34 +13,69 @@ import (
 )
 
 // IsJMXConfig checks if a certain YAML config is a JMX config
-func IsJMXConfig(name string, initConf integration.Data) bool {
+func IsJMXConfig(config integration.Config) bool {
+	for _, instance := range config.Instances {
+		isJMX := IsJMXInstance(config.Name, instance, config.InitConfig)
+		if isJMX {
+			return true
+		}
+	}
+
+	return false
+}
 
+// IsJMXInstance checks if a certain YAML instance is a JMX config
+func IsJMXInstance(name string, instance integration.Data, initConfig integration.Data) bool {
 	if _, ok := agentconfig.StandardJMXIntegrations[name]; ok {
 		return true
 	}
 
+	rawInstance := integration.RawMap{}
+	err := yaml.Unmarshal(instance, &rawInstance)
+	if err != nil {
+		return false
+	}
+
+	x, ok := rawInstance["is_jmx"]
+	if !ok {
+		return false
```

Comment:
You just want to continue testing here, but going directly to the init config params! Same for 2 next returns.